### ISSUE TITLE
crypto_sign_final_create() produces the wrong result?

### ISSUE THREAD

**Issue by jcayzac** [jedisct1/libsodium#896]
I'm trying to reproduce the result from [the official Ed25519ph test vectors]( https://tools.ietf.org/html/rfc8032#section-7.3):

```c
// SK and message taken from Ed25519ph test vectors at
// https://tools.ietf.org/html/rfc8032#section-7.3

crypto_sign_state state;
crypto_sign_init(&state);
crypto_sign_update(&state, (const unsigned char *) "\x61\x62\x63", 3);

unsigned char signature[64];
crypto_sign_final_create(&state, sig, 0, (const unsigned char *) "833fe62409237b9d62ec77587520911e9a759cec1d19755b7da901b96dca3d42");
for (int i=0; i<64; ++i) printf("%02x", sig[i]);
printf("\n");

//   Output: d3c601adc2e867d0a8b19d0f2cd2a6d00b346cb00659ac15d6df92387e32390ec9fe1a8f466a9e468100afc47bf6af090a883ff5edecd1b9b26ac5428b694003
// Expected: 98a70222f0b8121aa9d30f813d683f809e462b469c7ff87639499bb94e6dae4131f85042463c2a355a2003d062adf5aaa10b8c61e636062aaad11c2a26083406
```

I used libsodium 1.0.18. System is macOS 10.15.1.

**Comment by jcayzac** [comment]
Looking at the code, what `crypto_sign_final_create()` computes is `Ed25519(sha512(message))`. The documentation should stop mentioning Ed25519ph, which is a different thing.

**Comment by jedisct1** [comment]
The fact that you added a `(const unsigned char *)` cast because you probably originally saw the compiler complain about the wrong type being used should give you a hint about what is wrong with your code :)

**Comment by jcayzac** [comment]
@jedisct1 uhh… stupid me 😱 

I get the expected output when I decode the buffer. I think I got confused by the javascript bindings, which were throwing an invalid length error at me when I used a 32-byte buffer for the key. Thanks!

…but now I'm curious! [The code clearly just does `ED25519(SHA-512(message), pk)`](https://github.com/jedisct1/libsodium/blob/927dfe8e2eaa86160d3ba12a7e3258fbc322909c/src/libsodium/crypto_sign/ed25519/sign_ed25519.c#L61-L85), yet according to the RFC for Ed25519ph `dom2(x, y)` isn't the empty string as in pure Ed25519 but the string `SigEd25519 no Ed25519 collisions\1\0` (including the `\0` at the end, which is the length of the empty context).

Are the official test vectors wrong? I feel very confused now.
